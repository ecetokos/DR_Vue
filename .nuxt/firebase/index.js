import createApp from './app.js'

import authService from './service.auth.js'

const appConfig = {"apiKey":"AIzaSyD0c9a38G_8L7k09tqHmrmU2Hk_CeUBhEQ","authDomain":"dr-firebase-9de85.firebaseapp.com","projectId":"dr-firebase-9de85","storageBucket":"dr-firebase-9de85.appspot.com","messagingSenderId":"602632802158","appId":"1:602632802158:web:f59e6d568054f7f743fb7f"}

export default async (ctx, inject) => {
  /****************************************
  **************** LAZY MODE **************
  ****************************************/

  const { firebase, session } = await createApp(appConfig, ctx)

  let servicePromises = []

  if (process.server) {
    servicePromises = [
      authService(session, firebase, ctx, inject),

    ]
  }

  if (process.client) {
    servicePromises = [
      authService(session, firebase, ctx, inject),

    ]
  }

  const [
    auth
  ] = await Promise.all(servicePromises)

  const fire = {
    auth: auth
  }

    inject('fireModule', firebase)
    ctx.$fireModule = firebase

  inject('fire', fire)
  ctx.$fire = fire
}
