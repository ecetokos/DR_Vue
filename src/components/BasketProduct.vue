<template>
<div class="header-new">
    <div class="head-menu"><div>{{basketitems}}</div>
        <div class="head-cart active">  
            <div v-if="denemequan==0" style="text-align: center;">
                Sepetinizde ürün bulunmamaktadır.
            </div>        
       <div v-else>    
     <ul id="products">
         <li v-for="(product,index) in basketproducts" v-bind:key="product" >
             <figure class="foto">
                 <a href="/Kitap/Son-Curet/Yilmaz-Ozdil/Arastirma-Tarih/Politika-Arastirma/Turkiye-Politika-/urunno=0001889645001">
                    <img :src="product.img" style="height:85px" alt="Son Cüret"></a>
             </figure>
             <div class="detail">
                 <p class="product-name"><a title="Son Cüret" href="/Kitap/Son-Curet/Yilmaz-Ozdil/Arastirma-Tarih/Politika-Arastirma/Turkiye-Politika-/urunno=0001889645001">{{product.name}}</a></p>
                 <p class="product-type"><a title="Son Cüret" href="/Kitap/Son-Curet/Yilmaz-Ozdil/Arastirma-Tarih/Politika-Arastirma/Turkiye-Politika-/urunno=0001889645001">Medya Tipi: {{product.kapak}} / {{product.quantity}} ADET</a></p>
                 <p class="product-price"><a title="Son Cüret" href="/Kitap/Son-Curet/Yilmaz-Ozdil/Arastirma-Tarih/Politika-Arastirma/Turkiye-Politika-/urunno=0001889645001" style="color:black">{{(product.price*product.quantity).toFixed(2)}} TL</a></p>
             </div>
             <span style="display:none" id="index">{{index}}</span>
         </li>
     </ul>
     <div class="sum">
         <p class="sum-detail" id="totalquantity">TOPLAM {{totalquantityy}} ÜRÜN</p>
         <span class="sum-price" id="totalprice">{{totalprice.toFixed(2)}} TL</span>                
     </div>
     </div> 
        <nuxt-link to="/Sepetim"><input class="btn red buton" value="SEPETE GİT" type="button"></nuxt-link>
</div>
    </div>
    
    
</div>
    
</template>

<script>
import store from '../../store/index'
  export default {
    name: 'BasketProduct',
    data(){
        return{
            basketproducts:[],
            basketproductss:[],
            totalquan:0,
            denemequan:0,
            totalquantityy:0,
            totalprice:0
        }
    },
    computed:{
        /*basketproducts(){
            return this.$store.getters.cartProducts
        },*/
        /*total(){
            return this.$store.getters.cartTotal
        },*/
        quantitytotal(){
            var quantotal=0
            //return this.$store.getters.totalQuantity
            if(this.authUser && this.basketproducts){                      
                //alert(this.basketitemss[0].quantity)
                //alert(this.basketproducts[0].quantity)
                //alert(this.basketproducts[i].quantity)
                for(var i=0;i<this.denemequan;i++){
                    this.totalquantity += this.basketproducts[i].quantity
                }
                /*this.basketproducts.forEach(function (product){
                    quantotal+=product.quantity
                })*/
                //return this.basketproducts.reduce((total,product)=>total+product.quantity,0)
                
                return quantotal
            }
            else{return quantotal}
        },
        authUser(){     
          return this.$store.getters.takeuser
       },
       basketitems(){ 
           var sayi=0
           var toplam=0
           if(this.authUser){
            this.$fire.database.ref('/users/'+this.authUser.uid+'/basketitems').on('value',(snapshot)=>{      
                this.basketproductss=snapshot.val()
                this.denemequan=Object.keys(this.basketproductss).length
                if(this.denemequan>0){
                    snapshot.forEach((element) => {
                        sayi+=element.val().quantity 
                        toplam+=element.val().quantity*element.val().price       
                    })
                    this.totalquantityy=sayi
                    this.totalprice=toplam
                } 
            })
            if(this.basketproductss){
              this.basketproducts=this.basketproductss
            }
           }  
       }
    }
  }
</script>

<style scoped lang="scss">
@import "../css/dynamic-banner.css";
@import "../css/dr-custom.css";
@import "../css/style.css";
.header-new{
    background-color: white;
}
.head-cart{
    border: 1px solid;
    border-radius: 2px;
}
.foto{
    float: left;
    margin-left: 10px;
    margin-top: 11px;
    margin-right: 10px;
}
li{
    height: 110px;
    border: solid 1px lightgrey;
    
}
.detail{
    margin-top: 15px;
}
.sum{
    text-align: center;
    align-content: center;
}
.head-cart{
    width: 320px;
    padding: 20px;
}
.buton{
    width: 275px;
    align-content: center;
    margin-top: 10px;
}

.sum-detail
{
    font-size: 10px;
    color: gray;
    margin-top: 5px;
}
.sum-price{
    font-size: 20px;
}
.product-type{
    font-size: 11px;
    font-weight: bold;
}
hr{
    margin-left: 120px;
}
</style>