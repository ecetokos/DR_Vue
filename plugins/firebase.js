/*import * as firebase from 'firebase/app';
import 'firebase/auth'*/
import firebase from '@firebase/app';
require('firebase/auth');
/*import firebase from 'firebase'
require('firebase/auth')*/
/*
<script src="https://www.gstatic.com/firebasejs/8.2.1/firebase-app.js"></script>
<script>
  // Your web app's Firebase configuration
  
  // Initialize Firebase
  
</script>*/

var firebaseConfig = {
    apiKey: "AIzaSyD0c9a38G_8L7k09tqHmrmU2Hk_CeUBhEQ",
    authDomain: "dr-firebase-9de85.firebaseapp.com",
    projectId: "dr-firebase-9de85",
    storageBucket: "dr-firebase-9de85.appspot.com",
    messagingSenderId: "602632802158",
    appId: "1:602632802158:web:f59e6d568054f7f743fb7f"
  };
  
  firebase.initializeApp(firebaseConfig);

/*let app =null;
if(!firebase.apps.length){
    app=firebase.initializeApp(firebaseConfig);
}
export default firebase;*/