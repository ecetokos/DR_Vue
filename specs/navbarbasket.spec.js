import { mount,shallowMount } from '@vue/test-utils'
import NavBar from '../src/components/Navbar.vue'
import BasketProduct from '../src/components/BasketProduct.vue'
import Store from '../store/index'
import Vue from 'vue'
import Vuex from 'vuex'
import * as $fire from 'firebase';
Vue.use(Vuex)

test('is basketproduct vueinstance',()=>{
    const store=new Vuex.Store(Store)
    const wrappernb=mount(BasketProduct,{store})
    expect(wrappernb.isVueInstance()).toBe(true)
})

test('does basket product contain tages',()=>{
    const store=new Vuex.Store(Store)
    const wrapperbp=mount(BasketProduct,{store,data:{denemequan:1}})
    //wrapperbp.setData({denemequan:1})
    expect(wrapperbp.contains('div')).toBe(true)
    /*expect(wrapperbp.contains('p')).toBe(true)
    expect(wrapperbp.contains('span')).toBe(true)
    expect(wrapperbp.contains('ul')).toBe(true)*/
})

test('has the expected html structure',()=>{
    const store=new Vuex.Store(Store)
    const wrapperbp=mount(BasketProduct,{store})
    expect(wrapperbp.element).toMatchSnapshot()
})

describe('BasketProduct',()=>{
    it('has functions',()=>{
        expect(typeof BasketProduct.data).toBe('function')
    })
})