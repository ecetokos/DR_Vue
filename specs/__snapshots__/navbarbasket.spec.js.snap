// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does basket product contain tages 1`] = `
<div
  class="header-new"
>
  <div
    class="head-menu"
  >
    <div>
      
    </div>
     
    <div
      class="head-cart active"
    >
      <div
        style="text-align: center;"
      >
        
                Sepetinizde ürün bulunmamaktadır.
            
      </div>
       
      <nuxt-link
        to="/Sepetim"
      >
        <input
          class="btn red buton"
          type="button"
          value="SEPETE GİT"
        />
      </nuxt-link>
    </div>
  </div>
</div>
`;

exports[`has the expected html structure 1`] = `
<div
  class="header-new"
>
  <div
    class="head-menu"
  >
    <div>
      
    </div>
     
    <div
      class="head-cart active"
    >
      <div
        style="text-align: center;"
      >
        
                Sepetinizde ürün bulunmamaktadır.
            
      </div>
       
      <nuxt-link
        to="/Sepetim"
      >
        <input
          class="btn red buton"
          type="button"
          value="SEPETE GİT"
        />
      </nuxt-link>
    </div>
  </div>
</div>
`;
